python/cudf/cudf/core/indexing_utils.py

```diff
@@ -0,0 +1,243 @@
+    columns), namely one of:
+
+    - A boolean mask of the same length as the frame in the given
+      dimension
+    - A scalar integer that indexes the frame
+    - An array-like of integers that index the frame
+    - A slice that indexes the frame
+
+    Integer and slice-based indexing follows usual Python conventions.
+
+    Parameters
+    ----------
+    key
+        The key to destructure
+    frame
+        DataFrame or Series to provide context
+
+    Returns
+    -------
+    tuple
+        Indexers with length equal to the dimension of the frame
+
+    Raises
+    ------
+    IndexError
+        If there are too many indexers, or any individual indexer is a tuple.
+    """
+    n = len(frame.shape)
+    if isinstance(key, tuple):
+        # Key potentially indexes rows and columns, slice-expand to
+        # shape of frame
+        indexers = key + (slice(None),) * (n - len(key))
+        if len(indexers) > n:
+            raise IndexError(
+                f"Too many indexers: got {len(indexers)} expected {n}"
+            )
+    else:
+        # Key indexes rows, slice-expand to shape of frame
+        indexers = (key, *(slice(None),) * (n - 1))
+    indexers = tuple(k(frame) if callable(k) else k for k in indexers)
+    if any(isinstance(k, tuple) for k in indexers):
+        raise IndexError(
+            "Too many indexers: can't have nested tuples in iloc indexing"
+        )
+    return indexers
+
+
+def destructure_dataframe_iloc_indexer(
+    key: Any, frame: cudf.DataFrame
+) -> Tuple[Any, Tuple[bool, ColumnLabels]]:
+    """Destructure an index key for DataFrame iloc getitem.
+
+    Parameters
+    ----------
+    key
+        Key to destructure
+    frame
+        DataFrame to provide context context
+
+    Returns
+    -------
+    tuple
+        2-tuple of a key for the rows and tuple of
+        (column_index_is_scalar, column_names) for the columns
+
+    Raises
+    ------
+    TypeError
+        If the column indexer is invalid
+    IndexError
+        If the provided key does not destructure correctly
+    NotImplementedError
+        If the requested column indexer repeats columns
+    """
+    rows, cols = destructure_iloc_key(key, frame)
+    if cols is Ellipsis:
+        cols = slice(None)
+    scalar = is_integer(cols)
+    try:
+        column_names: ColumnLabels = list(
+            frame._data.get_labels_by_index(cols)
+        )
+        if len(set(column_names)) != len(column_names):
+            raise NotImplementedError(
+                "cudf DataFrames do not support repeated column names"
+            )
+    except TypeError:
+        raise TypeError(
+            "Column indices must be integers, slices, "
+            "or list-like of integers"
+        )
+    if scalar:
+        assert (
+            len(column_names) == 1
+        ), "Scalar column indexer should not produce more than one column"
+
+    return (rows, (scalar, column_names))
```

Comment:
Parens are extraneous here, I think.
```suggestion
    return rows, (scalar, column_names)
```